src/caretogether-pwa/src/Hooks/useAppNavigate.ts

```diff
@@ -30,6 +37,17 @@ export function useAppNavigate(): AppNavigate {
     dashboard: () => inContext(''),
     inbox: () => inContext('inbox'),
     family: (familyId: string) => inContext(`families/${familyId}`),
+    familyWithQuery: (
+      familyId: string,
+      referralId: string,
+      arrangementId?: string
+    ) =>
+      inContext(
+        `families/${familyId}?referralId=${referralId}${
+          arrangementId ? `&arrangementId=${arrangementId}` : ''
+        }`
+      ),
+
```

Comment:
You can use the same `family` function with some optional parameters:

```suggestion
    family: (familyId: string, referralId?: string, arrangement?: string) => {
      const searchParams = new URLSearchParams();

      if (referralId) {
        searchParams.append('referralId', referralId);
      }

      if (arrangement) {
        searchParams.append('arrangementId', arrangement);
      }

      const searchParamsString = searchParams.size
        ? `?${searchParams.toString()}`
        : '';

      return inContext(`families/${familyId}${searchParamsString}`);
    },

```